scripts/quests/sandoria/Spice_Gals.lua

```diff
@@ -0,0 +1,174 @@
+-----------------------------------
+-- Spice Gals
+-----------------------------------
+-- Log ID: 0, Quest ID: 110
+-----------------------------------
+-- CoP 3-1             : !addmission 6 258
+-- Rouva               : !pos -16.194 2.101 11.805
+-- Riverne - Site #A01 : !pos -514.717 -6.684 -407.314
+-- Riverne - Site #B01 : !pos -517.137 0.094 689.199
+-----------------------------------
+local southsandyID = zones[xi.zone.SOUTHERN_SAN_DORIA]
+-----------------------------------
+
+local quest = Quest:new(xi.questLog.SANDORIA, xi.quest.id.sandoria.SPICE_GALS)
+
+quest.reward =
+{
+    item = xi.item.MIRATETES_MEMOIRS,
+    fame = 40,
+    fameArea = xi.fameArea.SANDORIA,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_AVAILABLE and
+            player:hasCompletedMission(xi.mission.log_id.COP, xi.mission.id.cop.ANCIENT_VOWS)
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Rouva'] = quest:progressEvent(724),
+
+            onEventFinish =
+            {
+                [724] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.SOUTHERN_SAN_DORIA] =
+        {
+            ['Rouva'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:hasKeyItem(xi.ki.RIVERNEWORT) then
+                        return quest:progressEvent(725)
+                    else
+                        return quest:event(728):oncePerZone()
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [725] = function(player, csid, option, npc)
+                    if quest:complete(player) then
```

Comment:
complete works for re questing as well, so no need for the added ogic of checking for free space that was used in if complete section